packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -447,12 +455,13 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		];
 
 		if (maxCount > 0) {
-			const executions = await this.find({
-				select: ['id'],
-				skip: maxCount,
-				take: 1,
-				order: { id: 'DESC' },
-			});
+			const executions = await this.createQueryBuilder('execution')
+				.select('execution.id')
+				.where('execution.id NOT IN ' + annotatedExecutionsSubQuery.getQuery())
```

Comment:
nit: Could move the `.getQuery()` part to where this is defined and then here

```suggestion
				.where(`execution.id NOT IN (${annotatedExecutionsSubQuery})`)
```